packages/radix-vue/src/Stepper/StepperItem.vue

```diff
@@ -0,0 +1,159 @@
+const kbd = useKbd()
+
+const { forwardRef, currentElement } = useForwardExpose()
+
+const rootContext = injectStepperRootContext()
+const stepperItems = computed(() => Array.from(rootContext.stepperItems.value))
+
+const titleId = useId(undefined, 'radix-vue-stepper-item-title')
+const descriptionId = useId(undefined, 'radix-vue-stepper-item-description')
+
+const itemState = computed(() => {
+  if (completed.value)
+    return 'completed'
+  if (rootContext.modelValue.value === step.value)
+    return 'active'
+  if (rootContext.modelValue.value! > step.value)
+    return 'completed'
+  return 'inactive'
+})
+
+const isFocusable = computed(() => {
+  if (disabled.value)
+    return false
+  if (rootContext.linear.value) {
+    return step.value <= rootContext.modelValue.value! || step.value === rootContext.modelValue.value! + 1
+  }
+  return true
+})
+
+function handleMouseDown(event: MouseEvent) {
+  if (disabled.value)
+    return
+  if (rootContext.linear.value) {
+    if (step.value <= rootContext.modelValue.value! || step.value === rootContext.modelValue.value! + 1) {
+      if (event.ctrlKey === false) {
+        rootContext.changeModelValue(step.value)
+        return
+      }
+    }
+  }
+  else {
+    if (event.ctrlKey === false) {
+      rootContext.changeModelValue(step.value)
+      return
+    }
+  }
+
+  // prevent focus to avoid accidental activation
+  event.preventDefault()
+}
+
+function handleKeyDown(event: KeyboardEvent) {
+  event.preventDefault()
+
+  if (disabled.value)
+    return
+
+  if ((event.key === kbd.ENTER || event.key === kbd.SPACE) && !event.ctrlKey && !event.shiftKey) {
+    rootContext.changeModelValue(step.value)
+  }
+
+  if ([kbd.ARROW_LEFT, kbd.ARROW_RIGHT, kbd.ARROW_UP, kbd.ARROW_DOWN].includes(event.key)) {
+    useArrowNavigation(event, document.activeElement as HTMLElement, undefined, {
+      itemsArray: stepperItems.value,
+      focus: true,
+      loop: false,
+      arrowKeyOptions: rootContext.orientation.value,
+      dir: rootContext.dir.value,
+    })
+  }
+}
+
+onMounted(() => {
+  if (isFocusable.value)
+    rootContext.stepperItems.value.add(currentElement.value)
+})
+
+onUnmounted(() => {
+  rootContext.stepperItems.value.delete(currentElement.value)
+})
+
+watch(isFocusable, (newValue) => {
+  if (newValue)
+    rootContext.stepperItems.value.add(currentElement.value)
+  else
+    rootContext.stepperItems.value.delete(currentElement.value)
+})
+
+provideStepperItemContext({
+  titleId,
+  descriptionId,
+  state: itemState,
+  disabled,
+  step,
+})
+</script>
+
+<template>
+  <Primitive
+    :ref="forwardRef"
+    :as="as"
+    :as-child="asChild"
+    :aria-current="itemState === 'active' ? 'true' : undefined"
+    :data-state="itemState"
+    :disabled="disabled || !isFocusable ? '' : undefined"
+    :data-disabled="disabled || !isFocusable ? '' : undefined"
+    :data-orientation="rootContext.orientation.value"
+    :tabindex="isFocusable ? 0 : -1"
```

Comment:
Having `tabindex` on `StepperItem` looks kinda weird tho.. perhaps we can have a `button` element named `StepperTrigger` to allow user to focus, and trigger the `handleKeyDown`?

![image](https://github.com/radix-vue/radix-vue/assets/59365435/58887a89-1e20-4010-b7db-f68f509a701a)

Reply:
Sure thing! To clarify, the anatomy would be:
```
<StepperItem>
  <StepperTrigger>
      <StepperTitle />
      <StepperDescription />
   </StepperTrigger>
 <StepperSeparator />
 </StepperItem>
 ``` ?

Reply:
Yup I think this looks good 👍🏻